Mathlib/RingTheory/Bialgebra/MonoidAlgebra.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2025 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+import Mathlib.RingTheory.Bialgebra.Basic
+import Mathlib.RingTheory.Coalgebra.MonoidAlgebra
+
+/-!
+# The bialgebra structure on monoid algebras
+
+Given a monoid `X`, a commutative semiring `R` and an `R`-bialgebra `A`, this file collects results
+about the `R`-bialgebra instance on `A[X]` defined using `Finsupp.instCoalgebra R X A` from
+`Mathlib.RingTheory.Coalgebra.Basic`.
+
+## Main definitions
+
+* `(Add)MonoidAlgebra.instBialgebra`: the `R`-bialgebra structure on `A[X]` when `X` is an (add)
+  monoid and `A` is an `R`-bialgebra.
+* `LaurentPolynomial.instBialgebra`: the `R`-bialgebra structure on the Laurent polynomials
+  `A[T;T⁻¹]` when `A` is an `R`-bialgebra.
+-/
+
+suppress_compilation
+
+namespace MonoidAlgebra
+
+open Bialgebra
```

Comment:
I guess you could move this to the top of the file now, rather than repeating it.